src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
+import MenuIcon from "@mui/icons-material/Menu";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemText from "@mui/material/ListItemText";
+import Divider from "@mui/material/Divider";
+import Drawer from "@mui/material/Drawer";
+import LightMode from "@mui/icons-material/LightModeOutlined";
+import DarkMode from "@mui/icons-material/DarkModeOutlined";
+import { useTheme } from "@mui/material/styles";
+import { styled } from "@mui/material/styles";
+import { themeBreak } from "./ThemeHandler";
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+export const ColorModeContext = React.createContext({ toggleColorMode: () => {""} });
+// Desktop Nav
+const Responsive = styled("div")(() => ({
+    padding: themeBreak.spacing(1),
+    display: "none",
+    // Match [sm, md)
+    //       [600px, 900px)
+    [themeBreak.breakpoints.between("sm", "md")]: {
+        display: "flex",
+    },
+    [themeBreak.breakpoints.up("md")]: {
+        display: "flex",
+    },
+}));
+// Mobile Nav 
+const SmallView = styled("div")(() => ({
+    padding: themeBreak.spacing(1),
+    [themeBreak.breakpoints.down("sm")]: {
+        display: "flex",
+    },
+    [themeBreak.breakpoints.up("sm")]: {
+        display: "none",
+    },
+}));
 
-const TopNav = () => {
+interface Props {
+    /**
+     * Injected by the documentation to work in an iframe.
+     * You won"t need it on your project.
+     */
+    window?: () => Window;
+}
+export default function TopNav (props: Props) {
+    // Get Current MUI Theme    
+    const theme = useTheme();   
+    const colorMode = React.useContext(ColorModeContext); 
+    const LinkColorSwitch = (
+        theme.palette.mode === "dark" ? (
+            "#F3FFDE"
+        ) : (
+            "#191919"
+        )
+    )
+    
+    const Links = {
+        color: LinkColorSwitch,
+        textDecoration: "none",
+        fontWeight: 700,
+    };
+
+    const { window } = props;
+    const [mobileOpen, setMobileOpen] = React.useState(false);
+
+    const handleDrawerToggle = () => {
+        setMobileOpen(!mobileOpen);
+    };
+    // Theme Switch Icon Changes Based on Theme
+    const toggleButton = (
+        <IconButton
+            onClick={colorMode.toggleColorMode}
+            color="inherit"
+        >
+            {theme.palette.mode === "dark" ? (
+                <LightMode />
+            ) : (
+                <DarkMode />
+            )}
+        </IconButton>
+    );
+    // Hamburger Menu Content
+    const drawer = (
+        <Box onClick={handleDrawerToggle} sx={style.textCenter}>
+            <Typography variant="h6" sx={style.margin.drawerTitle}>
+                DevPrep
+            </Typography>
+            <Divider />
+            <List>
+                <ListItem disablePadding>
```

Comment:
These list items could also be separated into a separate component to reduce the need to retype the ListItem tags so many times, This would greatly reduce the amount of code in the component too. eg. in this component we have:
```
<BurgerLink to="/posts/general" text="General" />
```
in BurgerLink component we have:
```
<ListItem disablePadding>
    <ListItemButton sx={style.textCenter}>
        <ListItemText
            primary={
                <Link style={Links} to={ to }>
                    { text }
                </Link>
            }
        />
    </ListItemButton>
</ListItem>
```